File: frontend/src/types/skillpost.ts
export interface SkillPost {
  id: string;
  title: string;
  description: string;
  content: string;
  youtubeUrl?: string;
  userId: string;
  userName: string;
  tags: string[];
  createdAt: string;
  updatedAt: string;
  likes: number;
  likedBy: string[];
  comments: Comment[];
}

export interface Comment {
  id: string;
  content: string;
  userId: string;
  userName: string;
  createdAt: string;
  updatedAt?: string;
  parentCommentId?: string;
  replies?: Comment[];
}

export interface SkillPostRequest {
  title: string;
  description: string;
  content: string;
  youtubeUrl?: string;
  tags: string[];
}

export interface CommentRequest {
  content: string;
  parentCommentId?: string;
}

export interface SkillPostResponse {
  content: SkillPost[];
  totalElements: number;
  totalPages: number;
  number: number;
  size: number;
} 

File: backend/src/main/java/com/paf/knowledgenest/model/skillpost/SkillPost.java
package com.paf.knowledgenest.model.skillpost;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Set;
import java.util.HashSet;

@Data
@Document(collection = "skillposts")
public class SkillPost {

    @Id
    private String id;
    private String title;
    private String description;
    private String content;
    private String youtubeUrl;
    private String userId;
    private String userName;
    private List<String> tags = new ArrayList<>();
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private int likes = 0;
    private Set<String> likedBy = new HashSet<>();
    private List<Comment> comments = new ArrayList<>();
} 